facturacion_mexico/tests/test_issue162_clave_sat_obligatoria.py

```
@@ -0,0 +1,259 @@
+	def test_bloquea_cuando_un_item_sin_clave_sat(self):
+		"""ValidationError cuando al menos un ítem carece de clave SAT."""
+		si = _make_mock_si(
+			[
+				{"item_code": "ITEM-CON-CLAVE", "item_name": "Producto configurado"},
+				{"item_code": "ITEM-SIN-CLAVE", "item_name": "Producto sin configurar"},
+			]
+		)
+		clave_map = {"ITEM-CON-CLAVE": "43211501", "ITEM-SIN-CLAVE": None}
+
+		with patch("frappe.db.get_value", side_effect=self._make_get_value_mock(clave_map)):
+			with self.assertRaises(frappe.ValidationError) as ctx:
```

Ruff (0.15.15)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)